connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -811,12 +836,36 @@ private boolean startTask(ConnectorTaskId taskId) {
         );
     }
 
+    private Callable<Void> getTaskCallable(final ConnectorTaskId taskId, final TaskStatus.State targetState) {
+        return new Callable<Void>() {
+            @Override
+            public Void call() throws Exception {
+                switch (targetState) {
+                    case RUNNING:
+                        try {
+                            startTask(taskId);
+                        } catch (ConfigException e) {
```

Comment:
Again, there is an exception thrown when the `tasks` collection doesn't contain this task id. 

I was tempted to put everything within that try/catch block in the worker but I won't right now. For symmetry I'd keep all the try/catch blocks in start and stop in herder beside the method that stops a connector.